src/HasIdPrefix.php

```diff
@@ -14,7 +17,14 @@ private function __construct(string $value)
         $this->value = $value;
     }
 
-    abstract protected function failWhenPrefixIsInvalid(string $value): void;
+    abstract protected function prefix(): string;
+
+    protected function failWhenPrefixIsInvalid(string $value): void
```

Comment:
I'd make both this one and the abstract method `private`. We don't need them to be accessible in any inheritance chain.

Reply:
Abstract methods can't be private as they need to be defined in the extending class, but good eye!

Reply:
```Abstract private trait methods are only allowed since PHP 8.0```

Reply:
Wait, brainfart. Abstract method cannot be private in PHP 🙃